contracts/test/forkMainnet/AMMWrapper/Setup.t.sol

```diff
@@ -49,23 +47,34 @@ contract TestAMMWrapper is StrategySharedSetup {
     event SetFeeCollector(address newFeeCollector);
 
     // effectively a "beforeEach" block
-    function setUp() public virtual {
-        // Deploy and Setup Spender, AllowanceTarget, UserProxy, Tokenlon,
-        // PermanentStorage, ProxyPermanentStorage, AMMWrapper contracts
+    function setUp() public {
         setUpSystemContracts();
-        ammQuoter = new AMMQuoter(
-            UNISWAP_V2_ADDRESS,
-            UNISWAP_V3_ADDRESS,
-            UNISWAP_V3_QUOTER_ADDRESS,
-            SUSHISWAP_ADDRESS,
-            BALANCER_V2_ADDRESS,
-            IPermanentStorage(permanentStorage),
-            address(weth)
-        );
+        if (vm.envBool("DEPLOYED")) {
```

Comment:
Maybe these if/else code blocks in each strategy's `setUp` can be moved into a `_setUpDeployedStrategy` hook (just like the `_deployStrategyAndUpgrade` hook)?

Reply:
There are some setup or assignment which is related to strategy contract and cannot be combined together. For example not every strategy contract has the `feeCollector` assignment.

Reply:
I think they can be done as a hook like `_deployStrategyAndUpgrade`? `_deployStrategyAndUpgrade` is also strategy specific.

In `StrategySharedSetup.sol`
```solidity
function setUpSystemContracts() {
    if (vm.envBool("DEPLOYED")) {
        ...
        _setUpDeployedStrategy();
    } else {
        ...
    }
}

function _setUpDeployedStrategy() internal virtual returns (address) {}
```

In `RFQ.t.sol`
```solidity
function setUp() {
    ...
}

function _setUpDeployedStrategy internal override returns (address) {
    rfq = RFQ(payable(vm.envAddress("RFQ_ADDRESS")));
    owner = rfq.owner();
    feeCollector = rfq.feeCollector();
}
```

Reply:
Sounds good, adjusted in 551bf2fed94383cca8956ce924bb5c0bfa487b54.

But for the RFQ case, still have to overwrite tokens before `setUpSystemContracts()` when running in non-deployed environment. Do you think it should still be wrapped in a `_setupDeployedStrategy()` hook in RFQ or leave it in the setup if/else block?

Reply:
I think we can leave the token overwrite in `setUp` for now.

```solidity
function setUp() {
    if (!vm.envBool("DEPLOYED")) {
        // overwrite tokens with locally deployed mocks
        weth = IERC20(address(new MockWETH("Wrapped ETH", "WETH", 18)));
        ...
    }
    setUpSystemContracts();
    ...
}

function _setupDeployedStrategy() {
    rfq = RFQ(payable(vm.envAddress("RFQ_ADDRESS")));
    ...
}
```